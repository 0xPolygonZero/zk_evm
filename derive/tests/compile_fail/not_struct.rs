use zk_evm_derive::Columns;

#[repr(C)]
#[derive(Columns)]
enum Maybe<T> {
    Nothing,
    Just(T),
}

fn main() {}
