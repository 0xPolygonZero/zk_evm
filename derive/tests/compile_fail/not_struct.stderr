error: zk_evm_derive error: expected `struct`
 --> tests/compile_fail/not_struct.rs:3:1
  |
3 | / #[repr(C)]
4 | | #[derive(Columns)]
5 | | enum Maybe<T> {
6 | |     Nothing,
7 | |     Just(T),
8 | | }
  | |_^
